backend/src/test/java/com/crm/service/impl/AccountServiceImplTest.java

```diff
@@ -0,0 +1,278 @@
+import com.crm.util.EmailUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@RunWith(MockitoJUnitRunner.class)
+public class AccountServiceImplTest {
+
+    @Mock
+    private AccountRepository accountRepository;
+
+    @Mock
+    private RoleRepository roleRepository;
+
+    @Mock
+    private AccountMapper accountMapper;
+
+    @Mock
+    private EmailUtils emailUtils;
+
+    @Mock
+    private VerificationTokenServiceImpl verificationTokenService;
+
+    @Spy
+    private PasswordEncoder passwordEncoder;
+
+    @InjectMocks
+    private AccountServiceImpl accountService;
+
+    private static AccountEntity accountEntity;
+    private static AccountEntity savedAccountEntity;
+    private static AccountEntity activatedAccountEntity;
+    private static AccountResponse accountResponse;
+    private static NewAccountRequest newAccountRequest;
+    private static MyEmail myEmail;
+    private static VerificationTokenEntity neverExpiredVerificationToken;
+    private static VerificationTokenEntity expiredExpiredVerificationToken;
+
+
+    private static final Long id = 1L;
+    private static final String login = "Test48";
+    private static final String email = "Test48";
+    private static final String password = "Test48";
+    private static final boolean notActivated = false;
+    private static final boolean activated = true;
+    private static final Timestamp lastFailedLogin = Timestamp.from(Instant.now());
+    private static final Timestamp registrationDate = Timestamp.from(Instant.now());
+    private static final Timestamp lastSuccessfulLogin = Timestamp.from(Instant.now());
+    private static final int loginAttempts = 1;
+    private static final RoleEntity role = RoleEntity.builder().id(1L).name("admin").build();
+    private static final String verificationToken = "token";
+    private static final LocalDateTime neverExpiredToken = LocalDateTime.of(2050, 1, 1, 1, 1, 1);
```

Comment:
By convention, constants are written in CAPITAL_LETTERS

Reply:
done